app/assets/javascripts/admin/addon/controllers/modals/admin-penalize-user.js

```diff
@@ -0,0 +1,167 @@
+import Controller from "@ember/controller";
+import { action } from "@ember/object";
+import { next } from "@ember/runloop";
+import { inject as service } from "@ember/service";
+import { isEmpty } from "@ember/utils";
+import discourseComputed from "discourse-common/utils/decorators";
+import { extractError } from "discourse/lib/ajax-error";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import I18n from "I18n";
+import { Promise } from "rsvp";
+
+export default Controller.extend(ModalFunctionality, {
+  dialog: service(),
+
+  loadingUser: false,
+  errorMessage: null,
+  penaltyType: null,
+  penalizeUntil: null,
+  reason: null,
+  message: null,
+  postId: null,
+  postAction: null,
+  postEdit: null,
+  user: null,
+  otherUserIds: null,
+  loading: false,
+  confirmClose: false,
+
+  onShow() {
+    this.setProperties({
+      loadingUser: true,
+      errorMessage: null,
+      penaltyType: null,
+      penalizeUntil: null,
+      reason: null,
+      message: null,
+      postId: null,
+      postAction: "delete",
+      postEdit: null,
+      user: null,
+      otherUserIds: [],
+      loading: false,
+      errorMessage: null,
+      reason: null,
+      message: null,
+      confirmClose: false,
+    });
+  },
+
+  finishedSetup() {
+    this.set("penalizeUntil", this.user?.next_penalty);
+  },
+
+  beforeClose() {
+    // prompt a confirmation if we have unsaved content
+    if (
+      !this.confirmClose &&
+      ((this.reason && this.reason.length > 1) ||
+        (this.message && this.message.length > 1))
+    ) {
```

Comment:
For a bit of readability tweak:

```suggestion
    if (this.confirmClose) {
      return;
    }

    if (
      (this.reason && this.reason.length > 1) ||
      (this.message && this.message.length > 1)
    ) {
```